tests/lib.rs

```diff
@@ -754,6 +767,12 @@ fn test_parse_error_bad_non_string_primitive() {
         1234E+5678,
         1234e-5678,
         1234E-5678,
+        12.34e5678,
+        1234.E5678,
+        .1234e+5678,
+        12.34E+5678,
+        1234.e-5678,
```

Comment:
And this should be updated to check for too many leading zeroes.

Reply:
done